### PR TITLE
Remove support for unbounded number of polling threads

### PR DESCRIPTION
**Pull request by daniel-beck**
The default is 10. Previously unbounded configs will be set to 10.

----

I always thought it was weird to not allow >100 except that 0 means infinite…

WDYT @jenkinsci/code-reviewers ?

### Proposed changelog entries

* RFE: Always limit number of SCM polling threads to between 10 and 100.

### Submitter checklist

- [n/a] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

File: core/src/main/java/hudson/triggers/SCMTrigger.java [hunk 1]
```diff
@@ -313,7 +319,7 @@ public void setPollingThreadCount(int n) {
 
         @Restricted(NoExternalUse.class)
         public boolean isPollingThreadCountOptionVisible() {
-            if (getPollingThreadCount() != 0) {
+            if (getPollingThreadCount() != 10) {
                 // this is a user who already configured the option
```
Comment (jglick) [inline core/src/main/java/hudson/triggers/SCMTrigger.java]: Why does this logic even exist again?

**Review by jglick — APPROVED** [review]

File: core/src/main/java/hudson/triggers/SCMTrigger.java (hunk 1 above)
Comment (daniel-beck) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: WDYM? Not showing an advanced option unless it's likely to be needed?

Comment (jglick) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: Yeah I am not sure why code like this needs to exist. The normal UI is an **Advanced…** button, and if the value has been edited from its default (which can be inferred automatically by normal databinding), an “edited” icon appears on it.

Comment (daniel-beck) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: There's no non-advanced UI though, so it would be just a section title and an Advanced button?

Comment (jglick) [reply core/src/main/java/hudson/triggers/SCMTrigger.java]: Sure.

To be clear, I am not trying to block this, just noting a related UI simplification we could make.

**Comment by daniel-beck** [comment]
Form validation needs to be improved a bit here.

**Review by oleg-nenashev — APPROVED** [review]
🐝  and @reviewbybees done

@daniel-beck Do you plan to write more tests or do you want it to be merged as is?

**Comment by daniel-beck** [comment]
@oleg-nenashev Tests would be nice to have, but not sure by when I have the time to do them. No strong opinion.

**Review by jglick — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
I am fine with merging as is

**Review by aheritier — APPROVED** [review]
+1000

**Pull request merged**